src/main/java/org/sonar/plugins/stash/issue/collector/StashCollector.java

```diff
@@ -99,150 +102,180 @@ public static StashPullRequest extractPullRequest(String project, String reposit
-
-                                    StashUser author = extractUser(objAuthor);
-
-                                    StashComment comment = new StashComment(lineCommentId, lineCommentMessage, path, destination, author, lineCommentVersion);
-                                    diff.addComment(comment);
-
-                                    // get the tasks linked to the current comment
-                                    JSONArray jsonTasks = (JSONArray) jsonLineComment.get("tasks");
-                                    if (jsonTasks != null) {
-                                      for (Object objTask : jsonTasks.toArray()) {
-                                        JSONObject jsonTask = (JSONObject) objTask;
-
-                                        comment.addTask(extractTask(jsonTask.toString()));
-                                      }
-                                    }
-                                  }
-                                }
-                              }
-                            }
-                          }
-                        }
-
-                        result.add(diff);
-                      }
-                    }
-                  }
-                }
-              }
-            }
+    if (jsonDiffs == null) { return null; }
+
+    // Let's call this for loop "objdiff_loop"
+    for (Object objDiff : jsonDiffs.toArray()) {
+
+      JSONObject jsonDiff = (JSONObject)objDiff;
+      // destination path in diff view
+      // if status of the file is deleted, destination == null
+      JSONObject destinationPath = (JSONObject)jsonDiff.get("destination");
+
+      if (destinationPath == null) { continue; } // Let's process the next item in "objdiff_loop"
+
+      String path = (String)destinationPath.get("toString");
+      JSONArray jsonHunks = (JSONArray)jsonDiff.get("hunks");
 
-            // Extract File Comments: this kind of comment will be attached to line 0
-            JSONArray jsonLineComments = (JSONArray) jsonDiff.get("fileComments");
-            if (jsonLineComments != null) {
+      if (jsonHunks == null) { continue; } // Let's process the next item in "objdiff_loop"
 
-              StashDiff initialDiff = new StashDiff(StashPlugin.CONTEXT_ISSUE_TYPE, path, 0, 0);
+      // Let's call this for loop "objhunk_loop"
+      for (Object objHunk : jsonHunks.toArray()) {
 
-              for (Object objLineComment : jsonLineComments.toArray()) {
-                JSONObject jsonLineComment = (JSONObject) objLineComment;
+        JSONObject jsonHunk    = (JSONObject)objHunk;
+        JSONArray jsonSegments = (JSONArray)jsonHunk.get("segments");
 
-                long lineCommentId = (long) jsonLineComment.get("id");
-                String lineCommentMessage = (String) jsonLineComment.get("text");
-                long lineCommentVersion = (long) jsonLineComment.get(VERSION);
+        if (jsonSegments == null) { continue; } // Let's process the next item in "objhunk_loop"

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)